html_css/v2/main-course/custom-properties.md

```diff
@@ -0,0 +1,205 @@
+
+~~~html
+<p class='cool-paragraph'>Lorem ipsum dolor sit amet.</p>
+
+<p class='exciting-paragraph'>Lorem ipsum dolor sit amet.</p>
+~~~
+~~~css
+:root {
+  --main-color: red;
+}
+
+.cool-paragraph {
+  color: var(--main-color);
+}
+
+.exciting-paragraph {
+  background-color: var(--main-color);
+}
+~~~
+
+By declaring our custom property on the `:root` selector in the example above, we can access it on *any* other valid selector within our CSS file, since any other selector would be considered a descendant of the `:root` selector.
+
+### Themes
+
+Beyond allowing us to access custom properties more globally, the `:root` selector gives us one way to add themes to our pages. Keep in mind that we're not covering every way to add theming in this lesson, as we just want to introduce you to a simple way that you can get a lot of use out of.
+
+~~~html
+<html class='dark'>
+<!-- our boilerplate... -->
+  <body>
+    <div class='themed-element'>Our themed text</div>
+  </body>
+</html>
+~~~
+~~~css
+:root.dark {
+  --color-background: rgb(18, 18, 18);
+  --color-text: rgb(255, 255, 255);
+}
+
+:root.light {
+  --color-background: rgb(255, 255, 255);
+  --color-text: rgb(18, 18, 18);
+}
+
+.themed-element {
+  background-color: var(--color-background);
+  color: var(--color-text);
+}
+~~~
+
+We just created some quick themes! So what is actually going on here? First, we added a `class` attribute on our `html` element in the HTML file so that we can have a more specific selector to add custom properties onto, and so that our page has a default theme.
+
+In our CSS file we're just creating two scopes for our custom properties on the `:root` selector, one for when our `html` (or root) element has a class of `dark`, and another for when it has a class of `light`. Then our `.themed-element` selector will use the values of the custom properties depending on which class is currently present on our root element.
+
+Besides just adding custom properties to the CSS file, we need to add in some JavaScript to handle changing the theme, as well as some interactive element for the user to actually change it (such as a toggle or a switch). At this point, we trust that you can easily find out how to do both of these since that's a little out of scope for this lesson.
+
+#### Media Queries
+
+Giving users the ability to toggle a theme themselves is great, but there's another option for setting a theme that you may have noticed on certain sites or applications: using the user's theme setting from their operating system or user agent (like a browser).
+
+Instead of adding custom properties to the `:root` element depending on which class it has, we would add our custom properties on the `:root` element, without any class, inside of a `prefers-color-scheme` media query. Let's tweak our previous example a little bit to use this media query:
+
+~~~css
```

Comment:
i think best practice here is to first define a default theme outside the media-query and then put the other theme inside a media-query.

this will avoid any issues that might arise if `prefers-color-scheme` is unset, has a different value, or if the browser does not support it.

Reply:
So, one example I came across that fit your description here (others I came across just used two of these queries for both dark and light), mentioned that the `light` value is both used for when "light" is the preference *and* when there is no preference/an umatched preference. I tested this out on Ubuntu (putting the "dark" colors outside the query as the default and using the query with the "light" value) and it seems like this might be true, but the "Standard" theme might just be considered the light theme underneath it all.

I want to test this out a little more before comitting to changing this piece, but I do have it updated locally based on your suggestion for now.